### PR TITLE
feat: Migration of Development Environment and Testing tools

### PR DESCRIPTION
**Pull request by HoonBaek**
## For Internal Contributors

[UIKIT-2037](https://sendbird.atlassian.net/browse/UIKIT-2037)

## Description Of Changes

* Remove the `enzyme` and `react-test-renderer`
* Upgrade the `react` version to **v18**
* Upgrade the `storybook` version to **v6.5.10**
* Upgrade the `jest` and `babel-jest` to **v29**
* Upgrade the `jsdom` to **v20**
* Install `jest-environment-jsdom`
* Install `global-jsdom`
* Install `testing-library` (`@testing-library/react` and `@testing-library/jest-dom`)
* Migrate every tests with `testing-library` instead of the `enzyme` and `react-test-renderer`

## Types Of Changes

What types of changes does your code introduce to this project?
Put an `x` in the boxes that apply_

- [ ] Bugfix
- [ ] New feature
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance (ex) Prettier)
- [ ] Build configuration
- [x] Improvement (refactor code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ The CSS is in `./release/dist/index.css`
 
 ## Running the tests
 
-We have implemented tests for dumb ui components only. Technologies used: Jest, Enzyme and react-dom/test-utils
+We have implemented tests for dumb ui components only. Technologies used: Jest and testing-library
```
Comment (sravan-s) [inline README.md:84]: 👍

File: plop-templates/test.hbs [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
-import {shallow} from 'enzyme';
-import renderer from 'react-test-renderer';
+import { render, screen, fireEvent } from '@testing-library/reac';
```
Comment (sravan-s) [inline plop-templates/test.hbs:2]: 👍

File: src/ui/AdminMessage/__tests__/AdminMessage.spec.js [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
-import { shallow } from 'enzyme';
-import renderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
 
 import AdminMessage from "../index";
 import dummyAdminMessage from '../adminMessageDummyData.mock';
 
 describe('AdminMessage', () => {
-  it('should contain className', function () {
+  it.only('should contain className', function () {
```
Comment (sravan-s) [inline src/ui/AdminMessage/__tests__/AdminMessage.spec.js:8]: it.only should skip other tests, no? Any reason?

Comment (HoonBaek) [reply src/ui/AdminMessage/__tests__/AdminMessage.spec.js:8]: This is my mistake sorry and thanks

**Comment by sravan-s** [comment]
Do we really need `data-testid`?

**Comment by sicksid** [comment]
> Do we really need `data-testid`? ->
> Honestly, its good to add these, but, it could get more complex

@sravan-s it's widely used for the `testing-library` and `cypress`, makes it easier to test and find elements

**Comment by HoonBaek** [comment]
@sicksid 
We didn't set up our internal policy about `data-testid`
so let me remove them, and I will add them with a well-organized policy in the future

**Comment by HoonBaek** [comment]
We didn't set up our internal policy about `data-testid`.
`data-testid` attribute is good, but the value of `data-testid`s need to be a bit cleaner(follow a  naming convention)

let me remove them for now, and we will add them soon ~

**Review by sravan-s — APPROVED** [review]
Remove data-testid for now, LGTM

**Pull request merged**